centreon/www/class/centreonService.class.php

```diff
@@ -1691,37 +1694,38 @@ public function getCommandArgs($argArray = array(), $conf = array())
      * Returns service details
      *
      * @param int $id
+     * @param array $parameters
+     * @param boolean $monitoringDB
+     *
      * @return array
      */
-    public function getParameters($id, $parameters = array(), $monitoringDB = false)
+    public function getParameters($id, $parameters = [], $monitoringDB = false)
     {
-        $sElement = "*";
-        $arr = array();
-        if (empty($id)) {
-            return array();
-        }
-        if (count($parameters) > 0) {
-            $sElement = implode(",", $parameters);
-        }
-
-        $table = 'service';
-        $db = $this->db;
-        if ($monitoringDB) {
-            $table = 'services';
-            $db = $this->dbMon;
-        }
-
-        $res = $db->query(
-            "SELECT " . $sElement . " "
-            . "FROM " . $table . " "
-            . "WHERE service_id = " . $db->escape($id)
+        if ((int) $id <= 0) {
+            return [];
+        }
+
+        $searchColumns = (count($parameters) > 0) ? implode(',', $parameters) : '*';
+        $searchTable = ($monitoringDB === true) ? self::TABLE_SERVICE_REALTIME : self::TABLE_SERVICE_CONFIGURATION;
+        $database = ($monitoringDB === true) ? $this->dbMon : $this->db;
+
+        $statement = $database->prepare(<<<SQL
+            SELECT
+                $searchColumns
+            FROM
+                $searchTable
+            WHERE
+                service_id = :serviceId
+            SQL
         );
+        $statement->bindValue(':serviceId', (int) $id, \PDO::PARAM_INT);
+        $statement->execute();
 
-        if ($res->rowCount()) {
-            $arr = $res->fetchRow();
+        if ($statement->rowCount()) {
+            $result = $statement->fetchRow();
         }
 
-        return $arr;
+        return $result;
```

Comment:
`$result` can be undefined
```suggestion
        return [];
```